spec/controllers/users_controller_spec.rb

```diff
@@ -1,10 +1,41 @@
 require 'rails_helper'
 
+def login_user
+  @request.env['devise.mapping'] = Devise.mappings[:user]
+  sign_in FactoryBot.create(:user)
+end
+
 RSpec.describe UsersController, type: :controller do
-  describe 'GET #index' do
-    it 'returns http new_user_session_path' do
+  context 'GET #index without login' do
+    it 'redirect index' do
       get :index
       expect(response).to redirect_to new_user_session_path
     end
+    it 'when redirect index' do
+      get :index
+      expect(response).to have_http_status(302)
+    end
+    it 'current_user for not logged user' do
+      get :index
+      expect(subject.current_user).to eq(nil)
+    end
+  end
+  context 'GET #index with login' do
+    it 'when login user' do
+      login_user
+      get :index
+      expect(response).to have_http_status(200)
+    end
+
+    it 'current_user for logged user' do
+      login_user
+      get :index
+      expect(subject.current_user).to_not eq(nil)
+    end
+    it 'redirect to table' do
+      login_user
+      get :index
+      expect(request.env['PATH_INFO']).to eq('/')
```

Comment:
using `PATH_INFO` is wrong
don't use `request.env` object
Use rails methods for example `redirect_to`